clang/include/clang/Basic/DiagnosticParseKinds.td

```diff
@@ -1713,6 +1713,10 @@ let CategoryName = "Reflection Issue" in {
 def err_cannot_reflect_entity : Error<"cannot reflect entity">;
 def err_expansion_statements_disabled : Error<
   "'template for' statements are not enabled; use '-fexpansion-statements'">;
+
+def warn_parsing_ambiguity_in_refl_expression_with_ampamp_token : Warning <
+    "the compound condition may be misinterpreted due to '%0 &&' type parsing logic. did you mean '... (^%0) && ...'?"
+    >;
```

Comment:
IIRC, clang has a test which fails when you add a warning with no means of suppressing it (i.e., `-fWno-reflect-ambiguous-type` or whatever). We should fix that test before merging.

Also please constrain to 80 characters for stylistic consistency.

Reply:
Thank you for pointing out! That was the only one test that was failing on my laptop.

Fixed in 32c3567, re-run failed test + tested manually with `-Wno-reflect-parsing-ambiguity``